### PR TITLE
fixed cuda simulator break

### PR DESCRIPTION
**Pull request by rajgane07**

### CODE REVIEW

**Comment by rajgane07** [comment]
@rsln-s - I have done hot fix and all test cases passes. Suddenly upload to pypi test for wheel build is failing for windows, but source build succeeds . This is a weird error. I think this gh action issue. I will look into it. Probably this will go away after I run it sometime. Can you please check in GPU machine if the coda code works now?

**Comment by rsln-s** [comment]
@HaoTy, would you mind checking as well since you're the one who reported the issue?

**Comment by rsln-s** [comment]
Tests pass for me on a GPU instance.

**Review by HaoTy — COMMENTED** [review]
- GPU simulator's `get_expectation()` is not changed. 
- Avoid calling `np.asarray()` when the data is in GPU memory and there is subsequent computation done by the GPU. (The mpi `get_overlap()` numpy implementation is IMO not optimal, likely because cupy is not a hard dependency or an mpi-cuda native implementation is too complicated to worth the effort.)
- There is no need to multiply the entire energy vector by -1 when it can be done by a sign change to the final expectation value and `max()` instead of `min()` in overlap.
- Generally, I don't think these objective functions are good places to specify the "optimization type". It should be attached to each problem.

**Comment by rajgane07** [comment]
@HaoTy 

Can we have quick call on Monday? I do not have access to GPU instance, would like to see what you are seeing and what is expected to have better understanding of issues. The optimization type is passed from each problem and based on the problem the it is calculated for maximization and minimization.

**Comment by HaoTy** [comment]
> Can we have quick call on Monday?
Sure, just send me a Zoom link. I'm available after 2 pm Eastern time.

**Comment by rajgane07** [comment]
Let's meet at 2:30 PM EST today. Zoom - ruslan.s

**Comment by rajgane07** [comment]
@rsln-s @HaoTy 
I made the changes, I will try to test this in OmniAI by tomorrow. Can you also test from your side?

File: qokit/fur/nbcuda/qaoa_simulator.py [hunk 1]
```diff
@@ -119,11 +120,14 @@ def get_overlap(
                 costs_t = self._diag_from_costs(costs)
 
             if optimization_type == "max":
-                costs_t = -1 * np.asarray(costs_t)
+                costs_t = -1 * cp.asarray(costs_t)
```
Comment (HaoTy) [inline qokit/fur/nbcuda/qaoa_simulator.py:123]: This should be deleted after adding lines 126-130.

**Review by HaoTy — COMMENTED** [review]
Except for the line I commented, looks good to me. It's not very easy for me to test GPU and MPI as well, but I'd say they should work if the CPU version produces correct results. Let's leave where to put `optimization_type` for later.

**Comment by rajgane07** [comment]
@rsln-s @HaoTy 
I have resolved the issue and tested cuda simulator and python simulator in Omni AI. It gives me consistent results. 

Can you please test from your end? If everthing works, review and merge the code

**Comment by rsln-s** [comment]
The tests pass for me. @HaoTy, would you mind approving the PR with the latest changes if it looks good to you, or otherwise commenting on any outstanding issues?

**Pull request merged**